### PR TITLE
Add Python 3 and Django 1.8 support

### PR DESCRIPTION
**Pull request by grantmcconnaughey**

### CODE REVIEW

**Comment by grantmcconnaughey** [comment]
Huzzah! Finally got the build to pass. I added Django 1.8 support back. The only change was to the `hard_delete()` method. It now returns the number of results on Django 1.9 and simply returns None on Django 1.8.

I also added Python 3 support. Turns out this package already supported Python 3 just fine! I only had to change one test that referenced the built-in `long` (which is not in Python 3). I've updated the PyPI classifiers to indicate the versions of Python/Django this app supports. I also added them to the README so users can quickly see what is supported.

Lastly, I've updated the Travis build to test all of the supported versions of Python and Django, so if there are any build issues you'll know immediately what version is broken. This also required me to change the MySQL dependency to one that supports Python 3.

**Comment by grantmcconnaughey** [comment]
@adepue Any issues with this PR that I should address?

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
-  - DJANGO_VERSION_CEILING=1.9.1 DJANGO_DB_ENGINE=sqlite
-  - DJANGO_VERSION_CEILING=1.9.1 DJANGO_DB_ENGINE=postgres DJANGO_DB_NAME=testdb DJANGO_DB_USER=postgres
-  - DJANGO_VERSION_CEILING=1.9.1 DJANGO_DB_ENGINE=mysql DJANGO_DB_NAME=testdb DJANGO_DB_USER=travis
-  - DJANGO_VERSION_CEILING=1.10 DJANGO_DB_ENGINE=sqlite
-  - DJANGO_VERSION_CEILING=1.10 DJANGO_DB_ENGINE=postgres DJANGO_DB_NAME=testdb DJANGO_DB_USER=postgres
-  - DJANGO_VERSION_CEILING=1.10 DJANGO_DB_ENGINE=mysql DJANGO_DB_NAME=testdb DJANGO_DB_USER=travis
+  - DJANGO_VERSION=1.8.10 DJANGO_DB_ENGINE=sqlite
+  - DJANGO_VERSION=1.8.10 DJANGO_DB_ENGINE=postgres DJANGO_DB_NAME=testdb DJANGO_DB_USER=postgres
+  - DJANGO_VERSION=1.8.10 DJANGO_DB_ENGINE=mysql DJANGO_DB_NAME=testdb DJANGO_DB_USER=travis
+  - DJANGO_VERSION=1.9.3 DJANGO_DB_ENGINE=sqlite
```
Comment (adepue) [inline .travis.yml]: Can we leave these at 1.10 so we automatically test against new patch releases.  1.9.4 is already out.

File: .travis.yml [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
 env:
-  - DJANGO_VERSION_CEILING=1.9.1 DJANGO_DB_ENGINE=sqlite
-  - DJANGO_VERSION_CEILING=1.9.1 DJANGO_DB_ENGINE=postgres DJANGO_DB_NAME=testdb DJANGO_DB_USER=postgres
-  - DJANGO_VERSION_CEILING=1.9.1 DJANGO_DB_ENGINE=mysql DJANGO_DB_NAME=testdb DJANGO_DB_USER=travis
-  - DJANGO_VERSION_CEILING=1.10 DJANGO_DB_ENGINE=sqlite
-  - DJANGO_VERSION_CEILING=1.10 DJANGO_DB_ENGINE=postgres DJANGO_DB_NAME=testdb DJANGO_DB_USER=postgres
-  - DJANGO_VERSION_CEILING=1.10 DJANGO_DB_ENGINE=mysql DJANGO_DB_NAME=testdb DJANGO_DB_USER=travis
+  - DJANGO_VERSION=1.8.10 DJANGO_DB_ENGINE=sqlite
+  - DJANGO_VERSION=1.8.10 DJANGO_DB_ENGINE=postgres DJANGO_DB_NAME=testdb DJANGO_DB_USER=postgres
+  - DJANGO_VERSION=1.8.10 DJANGO_DB_ENGINE=mysql DJANGO_DB_NAME=testdb DJANGO_DB_USER=travis
```
Comment (adepue) [inline .travis.yml]: We can just make this 1.9 as well.

File: .travis.yml (hunk 1 above)
Comment (adepue) [inline .travis.yml]: Ah nevermind I see you moved this to not be a CEILING.  Any reason for that?

Comment (grantmcconnaughey) [inline .travis.yml]: I honestly don't remember. Let me fix it.

**Comment by grantmcconnaughey** [comment]
Reset Travis to use DJANGO_VERSION_CEILING. No idea why I changed it. It now tests against 1.8 and 1.9.

**Comment by adepue** [comment]
Thanks!  Merged, version bumped and pushed to pypi

**Pull request merged**